toonz/sources/toonz/xsheetviewer.cpp

```diff
@@ -911,6 +940,7 @@ void XsheetViewer::onSceneSwitched() {
 //-----------------------------------------------------------------------------
 
 void XsheetViewer::onXsheetChanged() {
+  updatePanelsSizes();
```

Comment:
For now you are notifying the toggle of Xsheet toolbar with the `TXsheetHandle::xsheetChanged` signal.
I think it would be better to use `TSceneHandle::preferenceChanged` signal instead, since the `xsheetChanged` will be emitted so often throughout common operations on the xsheet.
Please see my implementations for the "blank frame" setting in [here](https://github.com/opentoonz/opentoonz/blob/master/toonz/sources/toonz/preferencespopup.cpp#L372-L376) and [here](https://github.com/opentoonz/opentoonz/blob/master/toonz/sources/toonzqt/flipconsole.cpp#L1855-L1932).